File: Assets/Scripts/PlayerController.js
import UnityEngine;
import System;
class PlayerController extends MonoBehaviour {

	var speed = 800.0f; // can change this value in the inspector

	function FixedUpdate () { // infinite loop while in play mode; called before physics calculations for physics code
		var moveHorizontal : float = Input.GetAxis ("Horizontal"); // left-right input stored as a float between -1.0 and 1.0; x-value of Vector3
		var moveVertical : float = Input.GetAxis ("Vertical"); // down-up input stored as a float between -1.0 and 1.0; z-value of Vector3

		var movement = Vector3(moveHorizontal, 0.0f, moveVertical); // direction of the sum of both input values; 0.0f is y-value of Vector3 because don't want to move up

		GetComponent.<Rigidbody>().AddForce(movement * speed * Time.deltaTime); // applies force in desired direction; access rigidbody component
	}

}

File: Assets/Scripts/PlayerController.cs
using UnityEngine;
using System;
using UnityEngine.UI; // required to use "Text" object and other UI elements

public class PlayerController : MonoBehaviour {

	public float speed = 800.0f;
	public Text scoreText; // "scoreText" will store our UI Text object
	private int count = 0; // "count" will keep track of how many cubes we picked up
	public Text winText;

	void FixedUpdate () {
		float moveHorizontal = Input.GetAxis ("Horizontal");
		float moveVertical = Input.GetAxis ("Vertical");

		Vector3 movement = new Vector3 (moveHorizontal, 0.0f, moveVertical);

		GetComponent<Rigidbody>().AddForce (movement * speed * Time.deltaTime);
	}

	void OnTriggerEnter(Collider other) { // when the ball enters a Trigger Collider
		if (other.gameObject.tag == "PickUp") { // if the object that was entered has the tag "PickUp"
			other.gameObject.SetActive (false); // then set that object to be inactive (hide it)
			count += 1; // adds 1 to count each time we pick up a cube
			scoreText.text = "Score: " + count; // updates the text property of scoreText
		}
		if (count >=16) {
			winText.gameObject.SetActive(true);
		}
	}
}
